server/api/room/room.model.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+var mongoose = require('mongoose'),
+    Schema = mongoose.Schema;
+
+var RoomSchema = new Schema({
+    name: String,
```

Comment:
`name` and `location` fields should be required - see [mongo docs](https://mongoosejs.com/docs/validation.html) for more details (also rename `loc` to `location`